layer1/executor/tasks/base_task.go

```diff
@@ -80,7 +81,7 @@ func NewBaseTask(start uint64, end uint64, allowMultiExecution bool, subscribeOp
 }
 
 // Initialize default implementation for the ITask interface
-func (bt *BaseTask) Initialize(ctx context.Context, cancelFunc context.CancelFunc, database *db.Database, logger *logrus.Entry, eth layer1.Client, name string, id string, taskResponseChan TaskResponseChan) error {
+func (bt *BaseTask) Initialize(ctx context.Context, cancelFunc context.CancelFunc, database *db.Database, logger *logrus.Entry, eth layer1.Client, contracts layer1.AllSmartContracts, name string, id string, taskResponseChan TaskResponseChan) error {
```

Comment:
This is a problem in the current approach because the outside code creates the task objects and the scheduler initializes the parameters once the task is ready to be spawned (after the task was scheduled). May need some design refactoring, which we should discuss.